### ISSUE TITLE
[a11y] AnchorNav docs examples lack contrast in dark color mode

### ISSUE THREAD

**Issue by joshfarrant** [primer/brand#1126]
The text within the examples in the [AnchorNav docs](https://primer.style/brand/components/AnchorNav/) lack sufficient contrast when viewed in dark color mode.

<img width="970" height="1144" alt="Image" src="https://github.com/user-attachments/assets/d461a629-e8be-4597-a3e2-7a6faf92ac51" />
